deps/rabbit/src/rabbit_fifo_client.erl

```diff
@@ -732,10 +730,22 @@ maybe_auto_ack(false, {deliver, Tag, _Ack, Msgs} = Deliver, State0) ->
     {State, Actions} = settle(Tag, MsgIds, State0),
     {ok, State, [Deliver] ++ Actions}.
 
-
+handle_delivery(_Leader, {delivery, Tag, [_ | _] = IdMsgs},
+                #state{consumer_deliveries = CDels0} = State0) when not is_map_key(Tag, CDels0) ->
+    %% Note:
+    %% https://github.com/rabbitmq/rabbitmq-server/issues/3729
+    %%
+    %% Previously there had been a default value in the consumer deliveries map
+    %% (old code: maps:get(Tag, CDels0, #consumer{last_msg_id = -1}))
+    %% to cover cases where the consumer had been cancelled. We can't restore
+    %% the default value because the pending messages may not be delivered.
+    %% Instead, we choose to return the messages to the quorum queue process.
+    MsgIntIds = [Id || {Id, _} <- IdMsgs],
+    {State1, Deliveries} = return(Tag, MsgIntIds, State0),
+    {ok, State1, Deliveries};
 handle_delivery(Leader, {delivery, Tag, [{FstId, _} | _] = IdMsgs},
                 #state{cfg = #cfg{cluster_name = QName},
-                       consumer_deliveries = CDels0} = State0) ->
+                       consumer_deliveries = CDels0} = State0) when is_map_key(Tag, CDels0) ->
```

Comment:
As this is the common case I would put it before the other one. Also I do like a newline for the guards when they're wide. :)